Sources/SwiftGenKit/Parsers/Strings/StringsDictEntry.swift

```diff
@@ -0,0 +1,212 @@
+      variables.append(PluralEntry.Variable(key: variableKey, rule: variableRule))
+    }
+  }
+}
+
+// - MARK: Helpers
+
+extension StringsDict {
+  typealias VariableKeyResult = (key: String, range: Range<String.Index>, positionalArgument: Int?)
+
+  /// Parses variable keys and their ranges from a formatKey.
+  /// Every variable key that is contained within the format string is preceded
+  /// by the %#@ characters or by the positional variants, e.g. %1$#@, and followed by the @ character.
+  ///
+  /// - Parameter formatKey: The formatKey from which the variable keys should be parsed.
+  /// - Returns: An array of discovered variable keys and their range within the `formatKey`.
+  // swiftlint:disable:next discouraged_optional_collection
+  private static func variableKeysFromFormatKey(_ formatKey: String) -> [VariableKeyResult]? {
+    let pattern = #"(%(?>(\d)\$)?#@([\w\.\p{Pd}]+)@)"#
+    guard let regex = try? NSRegularExpression(pattern: pattern, options: []) else { return nil }
+    let nsrange = NSRange(formatKey.startIndex..<formatKey.endIndex, in: formatKey)
+    let matches = regex.matches(in: formatKey, options: [], range: nsrange)
+
+    return matches.compactMap { match -> (String, Range<String.Index>, Int?)? in
+      // 1st capture group is the whole format string including delimeters
+      let rangeNSRange = match.range(at: 1)
+      // 2nd capture group is the positional argument of the format string (Optional!)
+      let positionalArgumentNSRange = match.range(at: 2)
+      // 3rd capture group is the key, the string in between the delimeters
+      let keyNSRange = match.range(at: 3)
+
+      guard let rangeRange = Range(rangeNSRange, in: formatKey) else { return nil }
+      guard let keyRange = Range(keyNSRange, in: formatKey) else { return nil }
+      guard let positionalArgumentRange = Range(positionalArgumentNSRange, in: formatKey) else {
+        return (String(formatKey[keyRange]), rangeRange, nil)
+      }
+
+      return (String(formatKey[keyRange]), rangeRange, Int(formatKey[positionalArgumentRange]))
+    }
+  }
+}
+
+extension StringsDict.PluralEntry {
+  var translation: String? {
+    var result = formatKey
+
+    for intermediateResult in sequence(first: result, next: unfurlVariableKeysInFormatKey(_:)) {
+      result = intermediateResult
+    }
+
+    return "Plural case 'other': \(result)"
+  }
+
+  /// Unfurls any variable keys in a formatKey to get a possible translation for the `other` case.
+  ///
+  /// This method should be used in a recursive context, in which the output will be used as the input for the
+  /// next iteration.
+  ///
+  /// - Parameter formatKey: A format key containing variables.
+  /// - Returns: The format key in which all its variable keys are replaced with the content of the `other` translation,
+  ///  if the `formatKey` contained any variable keys, `nil` otherwise.
+  private func unfurlVariableKeysInFormatKey(_ formatKey: String) -> String? {
```

Comment:
Doesn't that recursive processing mean that we _do_ support recursive variables after all, contrary to what we stated in the README?

Reply:
This method is recursive, but it has nothing to do with the unsupported nested variables. We need to use recursion here, because we can only replace one variable at a time, because after that one replacement all ranges of the other variables become invalid. Therefore this method is being executed in the following manner:

```swift
unfurlVariableKeysInFormatKey("%#@files@ (%#@bytes@, %#@minutes@)")
// -> %d files remaining (%#@bytes@, %#@minutes@)

unfurlVariableKeysInFormatKey("%d files remaining (%#@bytes@, %#@minutes@)") 
// -> %d files remaining (%d bytes, %#@minutes@)

unfurlVariableKeysInFormatKey("%d files remaining (%d bytes, %#@minutes@)") 
// -> %d files remaining (%d bytes, %d minutes)
```

Reply:
I git that, but what if the variable `files` is itself defined with a nested variable for example? I think that your logic will unfurl that nested variable too due to the recursive nature of this code and will thus take it into account wouldn't it?

```

unfurlVariableKeysInFormatKey("%#@files@ (%#@bytes@, %#@minutes@)")
// -> %d %#@nestedVar@ remaining (%#@bytes@, %#@minutes@)
```